graphql/schemabuilder/pagination.go

```diff
@@ -767,18 +807,47 @@ func (c *connectionContext) consumeSorts(sb *schemaBuilder, m *method, typ refle
 	c.SortFields = make(map[string]*graphql.Field)
 
 	for name, fn := range m.SortFuncs {
-		sortableTyp := getFuncReturnType(fn)
-		if !supportedSort(sortableTyp) {
-			return fmt.Errorf(
-				"invalid sort field %s: unsupported return type %v, must be of kind int, uint, float or string",
-				name,
-				sortableTyp,
-			)
+		if fn.SortFunc != nil {
+			sortableTyp := getFuncReturnType(fn.SortFunc)
+			if !supportedSort(sortableTyp) {
+				return fmt.Errorf(
+					"invalid sort field %s: unsupported return type %v, must be of kind int, uint, float or string",
+					name,
+					sortableTyp,
+				)
+			}
+			c.SortFunctions[name] = getSort(sortableTyp)
+		}
+
+		if fn.BatchSortFunc != nil {
+			sortableTyp := getFuncReturnType(fn.BatchSortFunc)
+			if sortableTyp.Kind() != reflect.Map || !supportedSort(sortableTyp.Elem()) {
+				return fmt.Errorf(
+					"invalid sort field %s: unsupported return type %v, must be of kind int, uint, float or string",
+					name,
+					sortableTyp,
+				)
+			}
+			c.SortFunctions[name] = getSort(sortableTyp.Elem())
 		}
-		c.SortFunctions[name] = getSort(sortableTyp)
 
 		// Build a GraphQL field for the function.
-		field, err := sb.buildFunction(typ, &method{Fn: fn})
+		var field *graphql.Field = nil
+		var err error = nil
+		if fn.BatchSortFunc != nil && fn.SortFunc != nil {
+			field, err = sb.buildBatchFunction(typ,
```

Comment:
should we be using `buildBatchFunctionWithFallback` somewhere?

Reply:
good catch!